searchlib/src/vespa/searchlib/memoryindex/url_field_inverter.cpp

```diff
@@ -309,30 +307,29 @@ UrlFieldInverter::invertUrlField(const FieldValue &val)
             throw std::runtime_error(make_string("Expected URI struct, got '%s'", val.getDataType()->getName().c_str()));
         }
         break;
-    case CollectionType::WEIGHTEDSET:
-        if (cInfo.id() == WeightedSetFieldValue::classId) {
-            const auto &wset = static_cast<const WeightedSetFieldValue &>(val);
-            if (isUriType(wset.getNestedType())) {
-                processWeightedSetUrlField(wset);
-            } else {
-                throw std::runtime_error(make_string("Expected wset of URI struct, got '%s'", wset.getNestedType().getName().c_str()));
-            }
+    case CollectionType::WEIGHTEDSET: {
+        assert(val.isA(FieldValue::Type::WSET));
```

Comment:
Should these checks use `type() == ` instead, or conversely, should the `type() ==` checks earlier in this file use `isA()`?

Reply:
Fixed